pkg/security/securitycontextconstraints/provider.go

```diff
@@ -158,23 +158,24 @@ func (s *simpleProvider) CreatePodSecurityContext(pod *api.Pod) (*api.PodSecurit
 // Create a SecurityContext based on the given constraints.  If a setting is already set on the
 // container's security context then it will not be changed.  Validation should be used after
 // the context is created to ensure it complies with the required restrictions.
-func (s *simpleProvider) CreateContainerSecurityContext(pod *api.Pod, container *api.Container) (*api.SecurityContext, error) {
+func (s *simpleProvider) CreateContainerSecurityContext(pod *api.Pod, container *api.Container) (*api.SecurityContext, map[string]string, error) {
 	sc := securitycontext.NewEffectiveContainerSecurityContextMutator(
 		securitycontext.NewPodSecurityContextAccessor(pod.Spec.SecurityContext),
 		securitycontext.NewContainerSecurityContextMutator(container.SecurityContext),
 	)
+	annotations := maps.CopySS(pod.Annotations)
```

Comment:
do we need to copy this? we're copying top-level... we should be free to modify anything in the pod now

Reply:
If we don't need to copy, so why didn't you remove it -- https://github.com/kubernetes/kubernetes/pull/52849/commits/8c5b01376a4967dcda3650517bfa058fd26db8f5#diff-291b8dd7d08cc034975ddb3925dbb08fR245 ? Should we remove it then?

Reply:
oversight. we can remove it

Reply:
Ok, I've submitted https://github.com/kubernetes/kubernetes/pull/55504 for that.